llap-server/src/java/org/apache/hadoop/hive/llap/metrics/LlapDaemonExecutorMetrics.java

```diff
@@ -195,14 +202,22 @@ private LlapDaemonExecutorMetrics(String displayName, JvmMetrics jm, String sess
       this.executorThreadUserTime[i] = registry.newGauge(miu, 0L);
       this.executorNames.put(ContainerRunnerImpl.THREAD_NAME_FORMAT_PREFIX + i, i);
     }
+    if (simpleAverageWindowDataSize > 0) {
+      this.queueTime = new SynchronizedDescriptiveStatistics(simpleAverageWindowDataSize);
```

Comment:
took quick look at the implementation of `SynchronizedDescriptiveStatistics`  and i see synchronized everywhere,
maybe be this is not an issue yet but i can see how it can be become an issue in the future as a source of contentions, wondering if we can use a longAdder to compute ourself the the average, that should be lock free and will probably spread the contention.